src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -39,14 +39,16 @@ private Solution(SolutionState state)
         }
 
         internal Solution(Workspace workspace, SolutionInfo.SolutionAttributes solutionAttributes, SolutionOptionSet options, IReadOnlyList<AnalyzerReference> analyzerReferences)
-            : this(new SolutionState(workspace.PrimaryBranchId, workspace.Kind, workspace.Services, solutionAttributes, options, analyzerReferences))
+            : this(new SolutionState(workspace.PrimaryBranchId, workspace.Kind, workspace.PartialSemanticsEnabled, workspace.Services, solutionAttributes, options, analyzerReferences))
         {
         }
 
         internal SolutionState State => _state;
 
         internal int WorkspaceVersion => _state.WorkspaceVersion;
 
+        internal bool PartialSemanticsEnabled => _state.PartialSemanticsEnabled;
```

Comment:
>PartialSemanticsEnabled [](http://example.com/codeflow?start=22&length=23)

What is partial semantics?

Reply:
partial semantics are a way for features to say: give me a snapshot of whatever data you have right now, even if it's incomplete.  e.g. if you still have 1000 files to load/parse, i'd prefer to just pretend those files don't exist, rather than waiting on them.  It's for features that have extremely tight latency requirements (E.g. completion).

Generally speaking, this means during the start of VS you might observe some incomplete results (though it's rare to notice) but it means you don't have long delays on completion/commit.

Eventually, the workspace has caught up and you will then get the benefits of speed and completeness.  But this functionality allows features that care more about speed to let the workspace know.

Reply:
Why would we disable this per workspace?

Reply:
The general appraoch in the VS workspace is tehre is a 'bg compiler' that is cosntantly pulling on the data and bringing it up to completeness.  So if something asks for partial semantics, they know they're getting closer and closer to a 'complete' view, jsut that it will always be fast to request it.

in a workspace that doesn't have such a component running, then these components may continually see a *highly incomplete* view that never converges on completeness.  So this is an opt in systems for hosts that support this concept.  MSBuildWorkspace, for example, does not.

Reply:
I see. Would you mind adding these comments to the code?

Reply:
sure.  can i add in followup?